TerrorConsole/Assets/_Main/Scripts/Objects/Switchs/PuzzleSwitchLever.cs

```diff
@@ -0,0 +1,37 @@
+using UnityEngine;
+
+namespace TerrorConsole
+{
+    public class PuzzleSwitchLever : SwitchObject
+    {
+        private void OnCollisionEnter2D(Collision2D collision)
+        {
+            if (collision.gameObject.CompareTag("Player"))
+            {
+                InputManager.Source.OnActivateButton1 += AlternateState;
+            }
+        }
+
+        private void OnCollisionExit2D(Collision2D collision)
+        {
+            if (collision.gameObject.CompareTag("Player"))
+            {
+                InputManager.Source.OnActivateButton1 -= AlternateState;
+            }
+        }
+
+        protected override void AlternateState()
+        {
+            _state = !_state;
+            if (_state)
+            {
+                OnActivated?.Invoke(_id);
+                CameraSystemManager.Source.ShakeCamera();
+            }
+            else
+            {
+                OnDeactivated?.Invoke(_id);
+            }
+        }
```

Comment:
Move the events calling to `SwitchObject` and make the switch locked door optional.

Reply:
DM me in Discord for this, you will need to make `SwitchObject` more generic and it involves removing the locked door code.